messagingpush/src/main/java/io/customer/messagingpush/processor/PushMessageProcessorImpl.kt

```diff
@@ -0,0 +1,89 @@
+package io.customer.messagingpush.processor
+
+import android.content.Intent
+import androidx.annotation.VisibleForTesting
+import io.customer.messagingpush.MessagingPushModuleConfig
+import io.customer.messagingpush.util.PushTrackingUtil
+import io.customer.sdk.CustomerIOShared
+import io.customer.sdk.data.request.MetricEvent
+import io.customer.sdk.repository.TrackRepository
+
+internal class PushMessageProcessorImpl(
+    private val moduleConfig: MessagingPushModuleConfig,
+    private val trackRepository: TrackRepository
+) : PushMessageProcessor {
+
+    /**
+     * Responsible for storing and updating recent messages in queue atomically.
+     * Once this method is called, the current implementation marks the notification
+     * as processed and future calls for same [deliveryId] will receive true indicating
+     * that the notification is processed already.
+     *
+     * @param deliveryId unique message id received in push payload.
+     * @return true if the message was processed previously; false otherwise.
+     * Callers should generally process notifications only if false was returned.
+     */
+    @Synchronized
+    @VisibleForTesting
+    internal fun getOrUpdateMessageAlreadyProcessed(deliveryId: String?): Boolean {
+        val logger = CustomerIOShared.instance().diStaticGraph.logger
```

Comment:
maybe initialize it as class level, so other methods can use it too if needed?

Reply:
Updated